From 9bced58b8a5de1ff8022aee5d3d45e670ef3a1da Mon Sep 17 00:00:00 2001
From: Dan Liang <dan.liang@intel.com>
Date: Fri, 2 Aug 2013 15:18:09 +0800
Subject: libstagefright: add wmv mime type in ACodec

BZ: 127588

Allow wmv codec to be supported in ACodec.

Category: feature differentiation
Domain: Video.Media-video codecs
Origin: internal
Upstream-Candidate: No

Change-Id: I5d60bced806ce679e579935c5958302fa7bc56e0
Orig-Change-Id: Ie406ec906b583ebbb8c836fd2c259a85bd19710c
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/ACodec.cpp |   11 +++++++++++
 media/libstagefright/Utils.cpp  |   13 +++++++++++++
 2 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 1970789..05e5a10 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -43,6 +43,10 @@
 #define ROUND_16(X)     ((X + 0xF) & (~0xF))
 #endif
 
+#ifdef USE_INTEL_ASF_EXTRACTOR
+#include "MetaDataExt.h"
+#endif
+
 namespace android {
 
 template<class T>
@@ -1077,6 +1081,10 @@ status_t ACodec::setComponentRole(
             "video_decoder.vp8", "video_encoder.vp8" },
         { MEDIA_MIMETYPE_VIDEO_VP9,
             "video_decoder.vp9", "video_encoder.vp9" },
+#ifdef USE_INTEL_ASF_EXTRACTOR
+        { MEDIA_MIMETYPE_VIDEO_WMV,
+            "video_decoder.wmv", NULL },
+#endif
         { MEDIA_MIMETYPE_AUDIO_RAW,
             "audio_decoder.raw", "audio_encoder.raw" },
 #ifdef QTI_FLAC_DECODER
@@ -1812,6 +1820,9 @@ static const struct VideoCodingMapEntry {
     { MEDIA_MIMETYPE_VIDEO_MPEG2, OMX_VIDEO_CodingMPEG2 },
     { MEDIA_MIMETYPE_VIDEO_VP8, OMX_VIDEO_CodingVP8 },
     { MEDIA_MIMETYPE_VIDEO_VP9, OMX_VIDEO_CodingVP9 },
+#ifdef USE_INTEL_ASF_EXTRACTOR
+    {MEDIA_MIMETYPE_VIDEO_WMV, OMX_VIDEO_CodingWMV},
+#endif
 };
 
 static status_t GetVideoCodingTypeFromMime(
diff --git a/media/libstagefright/Utils.cpp b/media/libstagefright/Utils.cpp
index 9041c21..2350f0b 100644
--- a/media/libstagefright/Utils.cpp
+++ b/media/libstagefright/Utils.cpp
@@ -33,6 +33,10 @@
 #include "include/ExtendedUtils.h"
 #endif
 
+#ifdef USE_INTEL_ASF_EXTRACTOR
+#include "MetaDataExt.h"
+#endif
+
 namespace android {
 
 uint16_t U16_AT(const uint8_t *ptr) {
@@ -251,6 +255,15 @@ status_t convertMetaDataToMessage(
         buffer->meta()->setInt32("csd", true);
         buffer->meta()->setInt64("timeUs", 0);
         msg->setBuffer("csd-1", buffer);
+#ifdef USE_INTEL_ASF_EXTRACTOR
+    } else if (meta->findData(kKeyConfigData, &type, &data, &size)) {
+        sp<ABuffer> buffer = new ABuffer(size);
+        memcpy(buffer->data(), data, size);
+
+        buffer->meta()->setInt32("csd", true);
+        buffer->meta()->setInt64("timeUs", 0);
+        msg->setBuffer("csd-0", buffer);
+#endif
     }
 
 #ifdef QCOM_HARDWARE
-- 
1.7.4.1

